src/clarity_vm/clarity.rs

```diff
@@ -1908,25 +2099,84 @@ mod tests {
     #[test]
     pub fn test_block_limit() {
         let marf = MarfedKV::temporary();
-        let mut clarity_instance = ClarityInstance::new(false, marf, ExecutionCost::max_value());
+        let mut clarity_instance = ClarityInstance::new(false, marf);
         let contract_identifier = QualifiedContractIdentifier::local("foo").unwrap();
         let sender = StandardPrincipalData::transient().into();
 
+        pub struct BlockLimitBurnStateDB {}
+        impl BurnStateDB for BlockLimitBurnStateDB {
+            fn get_burn_block_height(&self, _sortition_id: &SortitionId) -> Option<u32> {
+                None
+            }
+
+            fn get_burn_header_hash(
+                &self,
+                _height: u32,
+                _sortition_id: &SortitionId,
+            ) -> Option<BurnchainHeaderHash> {
+                None
+            }
+
+            fn get_stacks_epoch(&self, _height: u32) -> Option<StacksEpoch> {
+                // Note: We return this StacksEpoch for every input, because this test is not exercising
+                // this method.
+                Some(StacksEpoch {
+                    epoch_id: StacksEpochId::Epoch20,
+                    start_height: 0,
+                    end_height: u64::MAX,
+                    block_limit: ExecutionCost {
+                        write_length: u64::MAX,
+                        write_count: u64::MAX,
+                        read_count: u64::MAX,
+                        read_length: u64::MAX,
+                        runtime: 100,
+                    },
+                    network_epoch: PEER_VERSION_EPOCH_2_0,
+                })
+            }
+
+            fn get_stacks_epoch_by_epoch_id(
+                &self,
+                _epoch_id: &StacksEpochId,
+            ) -> Option<StacksEpoch> {
+                self.get_stacks_epoch(0)
+            }
+            fn get_v1_unlock_height(&self) -> u32 {
+                u32::max_value()
+            }
+
+            fn get_pox_prepare_length(&self) -> u32 {
+                panic!("NullBurnStateDB should not return PoX info");
```

Comment:
This is `BlockLimitBurnStateDB`, not `NullBurnStateDB`.

Reply:
done